lib/delayed/job_wrapper.rb

```diff
@@ -4,8 +4,16 @@ class JobWrapper # rubocop:disable Betterment/ActiveJobPerformable
 
     delegate_missing_to :job
 
-    def initialize(job_data)
-      @job_data = job_data
+    def initialize(job_or_data)
+      # During enqueue the job instance is passed in directly, saves us deserializing
+      # it to find out how to queue the job.
+      # During load from the db, we get a data hash passed in so deserialize lazily.
+      if job_or_data.is_a?(ActiveJob::Base)
+        @job = job_or_data
+        @job_data = job.serialize
```

Comment:
Since we have the ActiveJob::Base already in a local variable, I think we can serialize that directly instead of round-tripping through the memoized `job` method.

```suggestion
        @job_data = job_or_data.serialize
```